lib/network/generate_key.go

```diff
@@ -0,0 +1,156 @@
+package sebaknetwork
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	host     = "localhost"
+	validFor = time.Duration(1000000)
+	rsaBits  = 2048
+)
+
+func publicKey(priv interface{}) interface{} {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &k.PublicKey
+	case *ecdsa.PrivateKey:
+		return &k.PublicKey
+	default:
+		return nil
+	}
+}
+
+func pemBlockForKey(priv interface{}) *pem.Block {
```

Comment:
You could accept `Signer` instead

Reply:
> You could accept Signer instead

You mean that type [crypto.Signer](https://golang.org/pkg/crypto/#PublicKey), right?
Please let me know more detail :)